### DISCUSSION TITLE
Need a smaller fingerprint from smiles

### DISCUSSION THREAD

**Discussion by phoenix-anna-naden** [rdkit/rdkit#7519] [FAQ]
I have this code
````
        ms = [Chem.MolFromSmiles(s) for s in smiles]
        fpgen = AllChem.GetRDKitFPGenerator()
        fps = [fpgen.GetFingerprint(x) for x in ms]
````
and what I need is a reasonably small representation of a molecule- like 128 floating point numbers. (I'm inputing it to the first layer of a neural network.

I was able to get an array of shape (1,2048) using the code
````
if __name__=='__main__':
        fps=smiles_to_ecfp(['COc1cncc(N)n1'])

        result=[np.frombuffer(bytes(fp.ToBitString(), 'utf-8'), 'u1') - ord('0') for fp in fps]
        print(np.array(result).shape)
````
But I don't know whether this is a good summary of the molecular structure and in any case, 2048 is too large

**Comment by Olabisi-Aishat-Bello** [comment]
You could try changing the parameter for fpSize to a smaller name like 128?

Basically doing AllChem.GetRDKitFPGenerator(fpSize = 128)

However, do note that reducing the array size leads to a higher risk of collisions due to the same bits being encoded to the same position which could not be useful for your neural network
